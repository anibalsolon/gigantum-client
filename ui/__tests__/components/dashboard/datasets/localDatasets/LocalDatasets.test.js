
      import React from 'react'
      import renderer from 'react-test-renderer';
      import {mount} from 'enzyme'
      import LocalDatasets from 'Components/dashboard/datasets/localDatasets/LocalDatasets';
      import json from './__relaydata__/LocalDatasets.json';
      import history from 'JS/history';
      import { Provider } from 'react-redux';
      import store from 'JS/redux/store';
      import { BrowserRouter as Router } from 'react-router-dom';

      import relayTestingUtils from '@gigantum/relay-testing-utils'


      const fixtures = {
        auth: () => {

        },
        datasetList: json.data.datasetList,
        localDatasets: json.data.datasetList,
        history,
        refetchSort: () => {

        },
        filterDatasets: data => data,
      };

      test('Test LocalDatasets', () => {

        const wrapper = renderer.create(
          relayTestingUtils.relayWrap(
            <Provider store={store}>
              <Router>
                <LocalDatasets
                    {...fixtures}
                />
              </Router>
            </Provider>, {}, json.data.localDatasets)
        );

        const tree = wrapper.toJSON()

        expect(tree).toMatchSnapshot()

      })
