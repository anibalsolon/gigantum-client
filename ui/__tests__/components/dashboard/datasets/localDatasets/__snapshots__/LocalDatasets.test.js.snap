// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test LocalDatasets 1`] = `
<div
  className="Datasets__listing"
>
  <div
    className="grid"
  >
    <div
      className="ImportModule Card Card--line-50 Card--text-center Card--add Card--import column-4-span-3"
    >
      <div
        className="Import__dataset-main"
      >
        <div
          className="Import__dataset-header"
        >
          <div
            className="Import__dataset-icon"
          >
            <div
              className="Import__dataset-add-icon"
            />
          </div>
          <div
            className="Import__dataset-title"
          >
            <h4>
              Add Dataset
            </h4>
          </div>
        </div>
        <div
          className="btn--import"
          onClick={[Function]}
        >
          Create New
        </div>
        <div
          className="btn--import"
          onClick={[Function]}
        >
          Import Existing
        </div>
        <div
          className="hidden createLabbook"
        >
          <div
            className="ToolTip__toggle createLabbook"
            onClick={[Function]}
          >
            <div
              className="ToolTip__glow-container"
            >
              <div
                className="ToolTip__glow-ring-outer"
              >
                <div
                  className="ToolTip__glow-ring-inner"
                />
              </div>
            </div>
          </div>
          <div
            className="hidden createLabbook"
          />
          <div
            className="hidden createLabbook"
          >
            Clicking on the 'Create New' button will open the new Project window, letting you name and select a Base for the new Project.
          </div>
        </div>
      </div>
      <div
        className="hidden"
      />
    </div>
    <a
      className="Card Card--text column-4-span-3 flex flex--column justify--space-between"
      href="/datasets/cbutler/mydataset"
      onClick={[Function]}
    >
      <div
        className="LocalDatasets__row--icons"
      />
      <div
        className="LocalDatasets__row--text"
      >
        <div>
          <h6
            className="LocalDatasets__panel-title"
            onClick={[Function]}
          >
            <span>
              <span
                className=""
              >
                mydataset
              </span>
            </span>
          </h6>
        </div>
        <p
          className="LocalDatasets__paragraph LocalDatasets__paragraph--owner "
        >
          cbutler
        </p>
        <p
          className="LocalDatasets__paragraph LocalDatasets__paragraph--owner"
        >
          Modified 25 days ago
        </p>
        <p
          className="LocalDatasets__paragraph LocalDatasets__paragraph--description"
        >
          <span />
        </p>
      </div>
      <div
        className="Tooltip LocalDatasetPanel__loading"
        data-tooltip="loading"
      />
    </a>
    <a
      className="Card Card--text column-4-span-3 flex flex--column justify--space-between"
      href="/datasets/cbutler/intursive-bodies-ireland"
      onClick={[Function]}
    >
      <div
        className="LocalDatasets__row--icons"
      />
      <div
        className="LocalDatasets__row--text"
      >
        <div>
          <h6
            className="LocalDatasets__panel-title"
            onClick={[Function]}
          >
            <span>
              <span
                className=""
              >
                intursive-bodies-ireland
              </span>
            </span>
          </h6>
        </div>
        <p
          className="LocalDatasets__paragraph LocalDatasets__paragraph--owner "
        >
          cbutler
        </p>
        <p
          className="LocalDatasets__paragraph LocalDatasets__paragraph--owner"
        >
          Modified 23 days ago
        </p>
        <p
          className="LocalDatasets__paragraph LocalDatasets__paragraph--description"
        >
          <span>
            <span
              className=""
            >
              Data on the offshore geologic intrusive bodies around Ireland. An intrusion is liquid rock that forms under Earth's surface. Magma from under the surface is slowly pushed up from deep within the earth into any cracks or spaces it can find.
            </span>
          </span>
        </p>
      </div>
      <div
        className="Tooltip LocalDatasetPanel__loading"
        data-tooltip="loading"
      />
    </a>
    <a
      className="Card Card--text column-4-span-3 flex flex--column justify--space-between"
      href="/datasets/dmk/ds2"
      onClick={[Function]}
    >
      <div
        className="LocalDatasets__row--icons"
      />
      <div
        className="LocalDatasets__row--text"
      >
        <div>
          <h6
            className="LocalDatasets__panel-title"
            onClick={[Function]}
          >
            <span>
              <span
                className=""
              >
                ds2
              </span>
            </span>
          </h6>
        </div>
        <p
          className="LocalDatasets__paragraph LocalDatasets__paragraph--owner "
        >
          dmk
        </p>
        <p
          className="LocalDatasets__paragraph LocalDatasets__paragraph--owner"
        >
          Modified 23 days ago
        </p>
        <p
          className="LocalDatasets__paragraph LocalDatasets__paragraph--description"
        >
          <span>
            <span
              className=""
            >
              dfghjfd
            </span>
          </span>
        </p>
      </div>
      <div
        className="Tooltip LocalDatasetPanel__loading"
        data-tooltip="loading"
      />
    </a>
    <a
      className="Card Card--text column-4-span-3 flex flex--column justify--space-between"
      href="/datasets/cbutler/data-file-zip-1"
      onClick={[Function]}
    >
      <div
        className="LocalDatasets__row--icons"
      />
      <div
        className="LocalDatasets__row--text"
      >
        <div>
          <h6
            className="LocalDatasets__panel-title"
            onClick={[Function]}
          >
            <span>
              <span
                className=""
              >
                data-file-zip-1
              </span>
            </span>
          </h6>
        </div>
        <p
          className="LocalDatasets__paragraph LocalDatasets__paragraph--owner "
        >
          cbutler
        </p>
        <p
          className="LocalDatasets__paragraph LocalDatasets__paragraph--owner"
        >
          Modified 22 days ago
        </p>
        <p
          className="LocalDatasets__paragraph LocalDatasets__paragraph--description"
        >
          <span />
        </p>
      </div>
      <div
        className="Tooltip LocalDatasetPanel__loading"
        data-tooltip="loading"
      />
    </a>
    <div
      className="DatasetPaginationLoader--hidden"
    >
      <div />
    </div>
    <div
      className="DatasetPaginationLoader--hidden"
    >
      <div />
    </div>
    <div
      className="DatasetPaginationLoader--hidden"
    >
      <div />
    </div>
    <div
      className="DatasetPaginationLoader--hidden"
    >
      <div />
    </div>
    <div
      className="DatasetPaginationLoader--hidden"
    >
      <div />
    </div>
  </div>
</div>
`;
